protocol/src/lib.rs

```diff
@@ -118,13 +118,15 @@ pub enum Message {
     HereIAm(u32, Uuid),
     YesItsMe(u32),
 
+    ReadOnlyMismatch(bool),
+
     /*
-     * Forcefully tell this downstairs to promote us (an Upstairs) to
-     * active.
+     * If read only is false, forcefully tell this downstairs to promote us
+     * (an Upstairs) to active. Kick out the old Upstairs.
      *
-     * Kick out the old Upstairs.
+     * If read only is true, multiple upstairs can connect ok.
      */
-    PromoteToActive(Uuid),
+    PromoteToActive(Uuid, bool),
```

Comment:
I wonder if the read_only needs to come from the protocol?  
Meaning, the downstairs knows if it is RO or not (and will be checking itself).  The downstairs does not care if the upstairs wants to promote to active with RO or not, it has the RO/RW setting already.  All the downstairs has to do is decide if it will honor the promote as RO or RW based on information it has locally.

Reply:
I figured it was the same as encrypted status - an Upstairs with an encryption context expects to only connect to a downstairs with the encrypted flag set to true, otherwise there's a `Encryption expectation mismatch!`.

If we're strict about it, only Upstairs that have read_only set can connect to downstairs with read_only set, and this would cause connections to fail during volume initialization instead of later as a response to user IO.